packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -304,6 +318,7 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	 * @returns the submitted commit. This is undefined if the underlying `SharedObject` is not attached,
 	 * and may differ from `commit` due to enrichments like detached tree refreshers.
 	 */
+	@breakingMethod
 	private submitCommit(
```

Comment:
I find it tempting to mark `submitCommit` as `@throwIfBroken` since we really don't want to be submitting if anything has gone wrong internally.  Are you concerned that we'd end up throwing a UsageError in this case (which is questionable since it is more indicative of an internal bug)?

Perhaps your intent with this PR is not to guard against such internal issues and only guard against outside calls. Is that it?